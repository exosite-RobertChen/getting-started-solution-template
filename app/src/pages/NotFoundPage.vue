<template>
  <v-container class="fill-height">
    <v-row class="whitevar--text" align="center" justify="center">
      <v-col class="text-center">
        <h1 class="mb-4">This page was not found</h1>
        <p>
          Error 404. The page you're looking for doesn't seem to be there...
        </p>
        <v-btn
          text
          color="primary"
          class="text-none"
          to="/"
          data-qa="HomePageLink"
          >Go to Home Page</v-btn
        >
      </v-col>
    </v-row>
  </v-container>
</template>
