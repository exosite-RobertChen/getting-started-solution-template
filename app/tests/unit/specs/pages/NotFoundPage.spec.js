import { shallowComponent } from '@/../tests/unit/common'
import NotFoundPage from '@/pages/NotFoundPage.vue'

const factory = overrides => shallowComponent(NotFoundPage, overrides)

describe('NotFoundPage.vue', () => {
  let wrapper = factory()
  test('can be mounted', () => {
    expect(wrapper.contains(NotFoundPage)).toBe(true)
  })

  test('matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  test('renders a go home button', () => {
    expect(wrapper.contains('[data-qa=HomePageLink]')).toBe(true)
  })
})
