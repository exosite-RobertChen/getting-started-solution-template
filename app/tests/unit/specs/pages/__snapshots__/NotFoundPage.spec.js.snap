// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage.vue matches snapshot 1`] = `
<v-container-stub
  class="fill-height"
  tag="div"
>
  <v-row-stub
    align="center"
    class="whitevar--text"
    justify="center"
    tag="div"
  >
    <v-col-stub
      class="text-center"
      tag="div"
    >
      <h1
        class="mb-4"
      >
        This page was not found
      </h1>
       
      <p>
        
        Error 404. The page you're looking for doesn't seem to be there...
      
      </p>
       
      <v-btn-stub
        activeclass=""
        class="text-none"
        color="primary"
        data-qa="HomePageLink"
        tag="button"
        text="true"
        to="/"
        type="button"
      >
        Go to Home Page
      </v-btn-stub>
    </v-col-stub>
  </v-row-stub>
</v-container-stub>
`;
